Game/Source/GDKTestGyms/Private/EventTracerComponent.cpp

```diff
@@ -0,0 +1,70 @@
+#include "EventTracerComponent.h"
+
+#include "Interop/Connection/SpatialEventTracerUserInterface.h"
+#include "Interop/Connection/SpatialTraceEventBuilder.h"
+#include "Interop/Connection/UserSpanId.h"
+#include "Net/UnrealNetwork.h"
+
+UEventTracerComponent::UEventTracerComponent()
+{
+	SetIsReplicatedByDefault(true);
+	PrimaryComponentTick.bCanEverTick = true;
+
+}
+
+void UEventTracerComponent::GetLifetimeReplicatedProps(TArray<FLifetimeProperty> & OutLifetimeProps) const
+{
+	Super::GetLifetimeReplicatedProps(OutLifetimeProps);
+
+	DOREPLIFETIME(UEventTracerComponent, bTestBool);
+}
+
+void UEventTracerComponent::BeginPlay()
+{
+	Super::BeginPlay();
+	if (OwnerHasAuthority() && bUseEventTracing)
+	{
+		GetWorld()->GetTimerManager().SetTimer(TimerHandle, this, &UEventTracerComponent::TimerFunction, 5.0f, true);
+	}
+}
+
+void UEventTracerComponent::TimerFunction()
+{
+	// Create a user trace event for sending a property update
+	FUserSpanId SpanId = USpatialEventTracerUserInterface::CreateSpanId(this);
+	USpatialEventTracerUserInterface::TraceEvent(this, SpanId, SpatialGDK::FSpatialTraceEventBuilder("user.send_component_property").GetEvent());
+	USpatialEventTracerUserInterface::AddLatentSpanId(this, this, SpanId);
+
+	bTestBool = !bTestBool;
+
+	// Create a user trace event for sending an RPC
+	SpanId = USpatialEventTracerUserInterface::CreateSpanId(this);
+	USpatialEventTracerUserInterface::TraceEvent(this, SpanId, SpatialGDK::FSpatialTraceEventBuilder("user.send_rpc").GetEvent());
+
+	USpatialEventTracerUserInterface::AddSpanIdToStack(this, SpanId);
```

Comment:
Could have a helper to run this in a lambda like how the delegate version works?